api/build.gradle

```diff
@@ -628,15 +684,15 @@ version = '0.1.0'          // Version in generated output
 sourceCompatibility = 1.8
 targetCompatibility = 1.8
 
-task incrementalHotSwap(type: IncrementalHotSwapTask) {
+task incrementalHotSwap(type: IncrementalHotSwapTask, dependsOn: 'compileJava') {
     inputDir = sourceSets.main.java.outputDir
 }
 
-incrementalHotSwap.dependsOn compileJava
-
 task listProjectAPIs(type: GenerateAPIListingTask)
 
 
+// TODO(jaycarlton) move tools-related stuff to its own Gradle project
```

Comment:
I don't disagree that this is probably the ideal end state, but I would not attempt to go down this path without first restructuring the codebase holistically to be more modular with cleaner dependencies. Originally, tools were a separate gradle file. This worked to a certain point and then completely broke down as more service dependencies were pulled in. In the end we wound up merging that into this file: https://precisionmedicineinitiative.atlassian.net/browse/RW-2172

Reply:
Definitely good to know. I'm pulling the TODOs out as they're mostly too prescriptive and the world may have changed before we get to them.